src/rng.c

```diff
@@ -38,7 +39,7 @@ void rng_init(void)
 			     sizeof("INTERNAL INSTANCE"));
 }
 
-void rng_gen(unsigned char *buf, size_t buflen)
+int rng_gen(unsigned char *buf, size_t buflen)
```

Comment:
The EC and ED key gens use this function, but do not check the return value.  But maybe first discuss with Joerg whether we want to fix this or remove the key gens and simply use OpenSSL.  This discussion is part of an ongoing effort to make libica FIPS 140-3 compliant.

Reply:
This commit also changes the implementation of `ica_[x|ed][25519|448]_key_gen()`, where `rng_gen()` is called. All these functions already return `-1` is `fips_check() != 0`. I just use not the same return value, if `rng_gen()` fails. This in in my opinion the correct solution and much better than doing a exit(1)!